draft-ietf-quic-transport.md

```diff
@@ -936,28 +936,86 @@ request - in response to the data carried in the early data from the client.
-perform source address validation during the handshake.  As long as the cookie
-cannot be successfully guessed by a client, the server can be assured that the
-client received the HelloRetryRequest.
-
-A server can use the HelloRetryRequest cookie in a stateless fashion by
-encrypting the state it needs to verify ownership of the client address and
-continue the handshake into the cookie.
-
-The ticket in the TLS NewSessionTicket message allows a server to provide a
-client with a similar sort of token.  When a client resumes a TLS connection -
-whether or not 0-RTT is attempted - it includes the ticket in the handshake
-message.  As with the HelloRetryRequest cookie, the server can include the state
-in the ticket it needs to validate that the client owns the address.
-
-A server can send a NewSessionTicket message at any time.  This allows it to
-update the state that is included in the ticket.  This might be done to refresh
-the ticket, or in response to changes in the state of a connection.
+Source address validation is therefore performed during the establishment of a
+connection.  TLS provides the tools that support the feature, but basic
+validation is performed by the core transport protocol.
+
+
+### Client Address Validation Procedure
+
+QUIC uses token-based address validation.  Any time the server wishes to
+validate a client address, it provides the client with a token.  As long as the
+token cannot be easily guessed (see {{token-integrity}}), if the client is able
+to return that token, it proves to the server that it received the token.
+
+During the processing of the cryptographic handshake messages from a client, TLS
+will request that the transport make a decision about whether to proceed based
+on the information it has.  TLS will provide the transport with any token that
+was provided by the client.  For an initial packet, the transport can decide to
+abort the connection, allow it to proceed, or request address validation.
+
+If the transport decides to request address validation, it provides the
+cryptographic handshake with a token.  The contents of this token are consumed
+by the server that generates the token, so there is no need for a single
+well-defined format.  A token could include information about the claimed client
+address (IP and port), and any other supplementary information the server will
+need to validate the token in the future.
+
+The cryptographic handshake is responsible for enacting validation by sending
+the address validation token to the client.  A legitimate client will include a
+copy of the token when they attempt to continue the handshake.  The
+cryptographic handshake extracts the token then asks the transport a second time
+whether the token is acceptable.  In response, the transport can either abort
+the connection or permit it to proceed.
+
+A connection MAY be accepted without address validation - or with only limited
```

Comment:
If you aren't going to send much data to the client, then it's a calculated risk.  As I observe here, there is implicit validation when the cryptographic handshake completes and a server might choose to avoid paying the tax of an explicit validation step (and an RTT) in favour of provisionally continuing.